src/west/commands/project.py

```diff
@@ -819,6 +819,16 @@ def _ref_ok(project, ref):
     return _git(project, 'show-ref --quiet --verify ' + ref, check=False) \
            .returncode == 0
 
+def _head_ok(project):
+    # Returns True if the reference 'HEAD' exists and is not a tag/branch.
+    # Some versions of git will report 1, when doing
+    # 'git show-ref --verify HEAD' even if HEAD is valid, see #119.
+    # 'git show-ref --head <reference>' will always return 0 if HEAD or
+    # <reference> is valid.
+    # We are only interested in HEAD, thus we must avoid <reference> being
+    # valid. Using a long random generated string should ensure this.
+    return _git(project, 'show-ref --quiet --head ' + str(os.urandom(64)), check=False) \
```

Comment:
Might get spaces and stuff from `os.urandom()`, which makes this a bit iffy. We could probably just hardcode something unlikely instead.

Wonder if `/` might be a guaranteed non-match.

Reply:
Maybe `git rev-parse` results on different Git versions could be mentioned too.

Reply:
> Might get spaces and stuff from `os.urandom()`, which makes this a bit iffy. We could probably just hardcode something unlikely instead.
> 
> Wonder if `/` might be a guaranteed non-match. It's not allowed to appear in refs.

yes, but when converting into `str`, you get something like 0x20, 0x1B, ...
which definitely should fail to match anything a human would start naming a tag or branch.

Good point with the '/'.
Although not allowed, it could be that it is possible, so will do some testing on that.

Reply:
> Maybe `git rev-parse` results on different Git versions could be mentioned too.

you mean `git show-ref` results, right ?

Reply:
> When converting into `str`, you get something like 0x20, 0x1B, ...
> which definitely should fail to match anything a human would start naming a tag or branch.

Oh, right, `os.urandom(64)` returns `bytes`, not strings.



> you mean `git show-ref` results, right ?

Was thinking `rev-parse` could be mentioned too, because that's what people usually use for this sort of stuff I think. Won't block for it though.

Reply:
@ulfalizer  `/` turned out to be allowed as argument to `show-ref` but it is not allowed as end-character on branches and tags, thus will not be able to get a match when used alone with `show-ref`.

Much nicer and simpler solution.